data/list/defs.lean

```diff
@@ -415,6 +415,13 @@ def map_last {α} (f : α → α) : list α → list α
 /- tfae: The Following (propositions) Are Equivalent -/
 def tfae (l : list Prop) : Prop := ∀ x ∈ l, ∀ y ∈ l, x ↔ y
 
+/-- `rotate l n` rotates the elements of `l` to the left by `n`
+  `rotate [0, 1, 2, 3, 4, 5] 2 = [2, 3, 4, 5, 0, 1] -/
+def rotate : list α → ℕ → list α
+| []     n     := []
+| l      0     := l
+| (a::l) (n+1) := rotate (l ++ [a]) n
```

Comment:
I'm not a fan of the quadratic algorithm here. The linear algorithm might be a bit less nice with the defeqs though, maybe call this one `rotate'`.

Reply:
the linear algorithm is:
```
let (l₁, l₂) := list.split_at (n % l.length) l in l₂ ++ l₁
```
although that traverses the list three times; there is a more elaborate one pass algorithm.

Reply:
Updated to faster definition